code/modules/recycling/sortingmachinery.dm

```diff
@@ -235,101 +255,173 @@
-				if(do_after(user,20, target = src))
-					if(!src || !W.isOn()) return
-					to_chat(user, "You sliced the floorweld off the delivery chute.")
-					var/obj/structure/disposalconstruct/C = new (src.loc)
-					C.ptype = 8 // 8 =  Delivery chute
-					C.update()
-					C.anchored = 1
-					C.density = 1
-					qdel(src)
-				return
-			else
-				to_chat(user, "You need more welding fuel to complete this task.")
+
+/obj/item/shippingPackage
+	name = "Shipping package"
+	desc = "A pre-labeled package for shipping an item to coworkers."
+	icon = 'icons/obj/storage.dmi'
+	icon_state = "shippack"
+	var/obj/item/wrapped = null
+	var/sortTag = 0
+	var/sealed = 0
+
+/obj/item/shippingPackage/attackby(obj/O, mob/user, params)
+	if(sealed)
+		if(istype(O, /obj/item/weapon/pen))
+			var/str = copytext(sanitize(input(user,"Intended recipient?","Address","")),1,MAX_NAME_LEN)
+			if(!str || !length(str))
+				to_chat(user, "<span class='notice'>Invalid text.</span>")
 				return
+			user.visible_message("<span class='notice'>[user] addresses [src] to [str].</span>")
+			name = "Shipping package (RE: [str])"
+		return
+	if(wrapped)
+		to_chat(user, "<span class='notice'>[src] already contains \a [wrapped].</span>")
+		return
+	if(istype(O, /obj/item) && !istype(O, /obj/item/weapon/storage) && !istype(O, /obj/item/shippingPackage))
+		var/obj/item/I = O
+		if(I.w_class > 3)
+			to_chat(user, "<span class='notice'>[I] is too large to fit in [src].</span>")
+		else
+			wrapped = I
+			user.unEquip(I)
+			I.forceMove(src)
+			I.add_fingerprint(usr)
+			add_fingerprint(usr)
+			to_chat(user, "<span class='notice'>You put [I] in [src].</span>")
+
+/obj/item/shippingPackage/attack_self(mob/user)
+	if(sealed)
+		to_chat(user, "<span class='notice'>You tear open [src], dropping the contents onto the floor.</span>")
+		playsound(loc, 'sound/items/poster_ripped.ogg', 50, 1)
+		user.unEquip(src)
+		wrapped.forceMove(get_turf(user))
+		wrapped = null
+		qdel(src)
+	else if(wrapped)
+		switch(alert("Select an action:",, "Remove Object", "Seal Package", "Cancel"))
+			if("Remove Object")
+				to_chat(user, "<span class='notice'>You shake out [src]'s contents onto the floor.</span>")
+				wrapped.forceMove(get_turf(user))
+				wrapped = null
+			if("Seal Package")
+				to_chat(user, "<span class='notice'>You seal [src], preparing it for delivery.</span>")
+				icon_state = "shippack_sealed"
+				sealed = 1
+				update_desc()
+	else
+		if(alert("Do you want to tear up the package?",, "Yes", "No") == "Yes")
+			to_chat(user, "<span class='notice'>You shred [src].</span>")
+			playsound(loc, 'sound/items/poster_ripped.ogg', 50, 1)
+			user.unEquip(src)
+			qdel(src)
+
+/obj/item/shippingPackage/proc/update_desc()
+	desc = "A pre-labeled package for shipping an item to coworkers."
+	if(sortTag)
+		desc += " The label says \"Deliver to [TAGGERLOCATIONS[sortTag]]\"."
+	if(!sealed)
+		desc += " The package is not sealed."
```

Comment:
Given how much you changed here, please comment at all the things you changed form the original code.